app/models/term.rb

```diff
@@ -36,6 +36,6 @@ def finish_previous
     end
 
     def make_active_previous
-      course.terms.last.update active: true if course.terms.any?
+      course.terms.last.update active: true if course.terms.count > 1
```

Comment:
Because there is error "cannot change frozen Hash" without this change.
This error cannot be detected if delete only one term, but promptly detected when trying to delete 2 or more terms.